Change onError(Exception) to onError(Throwable)

See issue "Observer#onError should use Throwable" for background => https://github.com/Netflix/RxJava/issues/296
